src/Bicep.Core/Registry/OciModuleRegistry.cs

```diff
@@ -94,6 +94,11 @@ public override bool IsArtifactRestoreRequired(OciModuleReference reference)
                 !this.FileResolver.FileExists(this.GetModuleFileUri(reference, ModuleFileType.Metadata));
         }
 
+        private InvalidModuleException GetUnknownArtifactException(string gotArtifactMediaType)
```

Comment:
Again, I had assumed the OCI code would be split into module vs provider.  Seems like this exception should be more explicit about what it wants.  If we're expecting a module artifact, we shouldn't be accepting a provider artifact, right?

Reply:
adressed